python_modules/dagster/dagster/_grpc/impl.py

```diff
@@ -401,24 +413,19 @@ def get_partition_config(
             partitioned_config,
         ) = _get_job_partitions_and_config_for_partition_set_name(repo_def, partition_set_name)
 
-        # Certain gRPC servers do not have access to the instance, so we only attempt to instantiate
-        # the instance when necessary for dynamic partitions: https://github.com/dagster-io/dagster/issues/12440
-        if _partitions_def_contains_dynamic_partitions_def(partitions_def):
-            with DagsterInstance.from_ref(
-                instance_ref
-            ) if instance_ref else nullcontext() as instance:
-                partition = partitions_def.get_partition(
-                    partition_key, dynamic_partitions_store=instance
-                )
-        else:
-            partition = partitions_def.get_partition(partition_key, dynamic_partitions_store=None)
+        with _instance_from_ref(instance_ref, partitions_def) as instance:
+            partition = partitions_def.get_partition(
```

Comment:
I think the `get_partition` call is unnecessary, cc @smackesey to confirm

I think we fetch the partition object within `get_run_config_for_partition_key` to validate that it is a valid partition

Reply:
I think you're right, but it's not returned, while we need the partition name out here for return in the `ExternalPartitionConfigData`.

Reply:
I think the partition key is always partition.name

Reply:
You're probably right but I think you should just remove it in your followup PR that already exists to avoid adding complexity and potential back-and-forth with @planvin, since these `get_partition` calls pre-existed his PR anyway.

Reply:
👍